### PR TITLE
Billing account module

### LINKED ISSUE #1721: New billing account module with budgets and sinks

### PR DESCRIPTION
**Pull request by ludoo**
Still missing

- [x] tests and examples
- [x] README

FAST IAM binding resources will be converted to the new module only if it improves legibility.

Fixes #1721

### CODE REVIEW

File: fast/stages/0-bootstrap/billing.tf [hunk 1]
```diff
@@ -74,12 +74,3 @@ resource "google_billing_account_iam_member" "billing_ext_admin" {
   role               = "roles/billing.admin"
   member             = each.key
 }
-
-resource "google_billing_account_iam_member" "billing_ext_cost_manager" {
```
Comment (juliocc) [inline fast/stages/0-bootstrap/billing.tf:78]: You don't assign this anywhere else. Is this no needed if you have `billing.admin`?

**Review by juliocc — APPROVED** [review]
very nice. Left a few minor comments here and there.

File: modules/billing-account/variables.tf [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    condition = alltrue(flatten([
+      for k, v in var.budgets : [
+        for kk, vv in v.update_rules : [
+          vv.monitoring_notification_channels != null
+          ||
+          vv.pubsub_topic != null
+        ]
+      ]
+    ]))
+    error_message = "Budget notification rules need either a pubsub topic or monitoring channels defined."
```
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/billing-account/variables.tf:117]: Monitoring 😀

File: modules/billing-account/README.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+          pubsub_topic = module.pubsub-billing-topic.id
+        }
+      }
+    }
+  }
+}
+# tftest modules=2 resources=2 inventory=budget-pubsub.yaml
+```
+
+#### Monitoring channels
```
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/billing-account/README.md:168]: Monitoring 😀

File: modules/billing-account/budgets.tf [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    projects = (
+      each.value.filter.projects == null
+      ? null
+      : each.value.filter.projects
+    )
+    resource_ancestors = (
+      each.value.filter.resource_ancestors == null
+      ? null
+      : each.value.filter.resource_ancestors
+    )
```
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/billing-account/budgets.tf:84]: Had a per.adifd without, but now that I look at it I'd makes little sense. Let me double check.

File: fast/stages/0-bootstrap/billing.tf (hunk 1 above)
Comment (ludoo) [reply fast/stages/0-bootstrap/billing.tf:78]: It overlaps with billing admin, yes

File: modules/billing-account/budgets.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/billing-account/budgets.tf]: of course, I got caught up troubelshooting an example and this is totally unneded :)

fixed

File: modules/billing-account/variables.tf [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  nullable = false
+  default  = {}
+  validation {
+    condition = alltrue([
+      for k, v in var.budgets : v.amount != null && (
+        try(v.amount.use_last_period, null) == true ||
+        try(v.amount.units, null) != null
+      )
+    ])
+    error_message = "Each budgets needs to have amount units specified, or use last period."
```
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/billing-account/variables.tf]: good chatch, fixed

File: modules/billing-account/budgets.tf [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_monitoring_notification_channel" "default" {
+  for_each    = var.budget_notification_channels
+  description = each.value.description
+  display_name = (
```
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/billing-account/budgets.tf]: of course...

fixed

**Pull request merged**